### ISSUE TITLE
MIssing plots in granger_fmri.html

### ISSUE THREAD

**Issue by justbennet** [nipy/nitime#173]
There might be something not right about the last two figures here

    http://nipy.org/nitime/examples/granger_fmri.html

Those are missing the body of the graph, which is all white.

I believe this is in the file `doc/examples/granger_fmri.py`.

**Comment by arokem** [comment]
Thanks for spotting this. Looks like a lot of nan's in the results from  Granger Causality calculation. Since it's been a while since the docs were built, I'll probably need to bisect back in history to see where this happened. Unfortunately, it might take me a little while to get to it.

**Comment by justbennet** [comment]
The documentation web site still has blank pages, and the HTTP headers show

$ wget -S http://nipy.org/nitime/examples/granger_fmri.html--2019-07-04
08:16:37--  http://nipy.org/nitime/examples/granger_fmri.html
Resolving nipy.org (nipy.org)... 185.199.110.153, 185.199.109.153,
185.199.111.153, ...
Connecting to nipy.org (nipy.org)|185.199.110.153|:80... connected.
HTTP request sent, awaiting response...
  HTTP/1.1 200 OK
  Server: GitHub.com
  Content-Type: text/html; charset=utf-8
  Last-Modified: Sat, 22 Jun 2019 22:01:11 GMT

Sorry for the reopen if the web page regeneration is done
automatically and I am just jumping the gun.

**Comment by arokem** [comment]
Yeah. I still haven't had time to look into this...

**Comment by justbennet** [comment]
Thanks.  Just curious because the issue got closed but the problem seems to remain...?  There is also a question on the Neuroimaging mailing list about it,

Subject: [Neuroimaging] GC analysis with nitime
Date: Fri, Jul 5, 2019 at 7:37 PM

Should the issue be reopened until it gets addressed?

**Comment by arokem** [comment]
Clearly.

**Comment by hvannieuwenh** [comment]
Hi, I had the same issue when trying to run the Granger causality example for fMRI on my own computer. I tried rolling back to older versions of nitime (0.7, 0.6, and 0.5) but they had their own problems with dependent packages. Is there any way on my end to make the Granger causality module functional? It is exactly the tool I am looking for in my research. Thanks.

**Comment by Xunius** [comment]
Hi all,
I spotted the same issue. Although being totally ignorant about the inner maths about Granger, I think I figured out where the problem lies.

The `GrangerAnalyzer` puts its outputs in the **upper** triangle:

```
        if ij is None:
            # The following gets the full list of combinations of
            # non-same i's and j's:
            x, y = np.meshgrid(np.arange(self._n_process),
                               np.arange(self._n_process))
            self.ij = list(zip(x[tril_indices_from(x, -1)],
                          y[tril_indices_from(y, -1)]))
```

Note the order is `x` then `y`, so `self.ij` indexes into the upper triangle.

While in making the plot, `drawmatrix_channels()` leaves only the **lower** triangle:

```
    # Null the upper triangle, so that you don't get the redundant and the
    # diagonal values:
    idx_null = triu_indices(m.shape[0])
    m[idx_null] = np.nan
```
So we ended up with nothing.

I suppose the fix is to transpose the ordering in `GrangerAnalyzer` like so:

```
           self.ij = list(zip(y[tril_indices_from(y, -1)],
                          x[tril_indices_from(x, -1)]))
```

This gives the following image for the `fig03` in the tutorial:

![Figure_3](https://user-images.githubusercontent.com/8076931/134790922-73a379f3-9b42-48b0-b4cf-6a48b6e77802.png)

`fig04` is like this:

![Figure_4](https://user-images.githubusercontent.com/8076931/134790933-3bcd0810-5365-45df-a480-cbd0688f9908.png)

But, I'm not familiar with the plotting conventions people use in such like of analyses. Do we assume the plot in `fig03` is the "causality" from row to column, or the other way around? Please help verity the proposed fix.

**Comment by arokem** [comment]
Thanks for figuring this out! I think that the best fix would probably be to alter the behavior of `drawmatrix_channels`. Do you want to try to make a PR for that?

**Comment by Xunius** [comment]
@arokem I'm not sure what exact changes to make to `drawmatrix_channels()`, because it's working fine for other plots and only fails for Granger. How to change it so we don't break compatibility with other tutorials and with older versions?

**Comment by arokem** [comment]
Gotcha. Yeah - I think that your original solution makes more sense then. Arguably, this is the design we should have implemented to begin with. Could you make a PR with that implemeted?

**Comment by Xunius** [comment]
@arokem It seems that the fix is failing the `test_granger.py` test. Now I actually think that proposed fix is not ideal, it would break with old behavior. We need to be more careful.

**Comment by arokem** [comment]
Yes - we need to adjust the test and warn about the break to backwards compatibility. 

If you reopen the PR, we can discuss that there.

**Comment by Xunius** [comment]
I made some back-n-forth changes, hopefully not creating too much confusion to the git history.
In the latest PR I changed the `drawmatrix_channels()` instead to use the **upper** triangle but transpose it lower before plotting. This doesn't affect other tutorials because coherence, correlation etc. are all symmetrical. I've tested all scripts in `examples` that calls `drawmatrix_channels()` and the figures all look good.
